### ISSUE TITLE
Unexpected EOF when creating Extension

### ISSUE THREAD

**Issue by HenryKeen** [labd/terraform-provider-commercetools#70]
Hi

I'm having an issue when registering an API extension.

I'm seeing a pretty generic error message related to the extension itself. I've turned on Debug logging and can't see any useful logs. I'm loading the accesskey/secret from Systems Manager.

My extension is defined as:
```
resource "commercetools_api_extension" "ct_extension" {
  key = "commerce-order-apiextension"

  destination {
    type          = "AWSLambda"
    arn           = "${aws_lambda_function.lambda.arn}"
    access_key    = "${aws_iam_access_key.ct_extension_user.id}"
    access_secret = "${aws_iam_access_key.ct_extension_user.secret}"
  }

  trigger {
    resource_type_id = "order"
    actions          = ["Create"]
  }
}
```

Error message:

```
Error: Error applying plan:

1 error(s) occurred:

* commercetools_api_extension.ct_extension: 1 error(s) occurred:

* commercetools_api_extension.ct_extension: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


2019/06/19 14:20:47 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-06-19T14:20:47.652Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
[terragrunt] 2019/06/19 14:20:47 Hit multiple errors:
exit status 1
```

I expect the error message doesn't mean much to you either so is there anything else I can be doing to debug this?

Thanks!

**Comment by davidweterings** [comment]
Hey @HenryKeen , there are two extra environment variables you can set to enable debug output. The first one is for Terraform: `TF_LOG=1`, for the Commercetools part you can enable `CTP_DEBUG=1`. Note this generates a *lot* of output but usually this allows you to figure out what is going on.

**Comment by mvantellingen** [comment]
Hi @HenryKeen,

Can you run terraform with `TF_LOG=1 CTP_DEBUG=1 terraform apply`. It should show a log of all the calls happening to the commercetools platform.

**Comment by HenryKeen** [comment]
Thanks for such a quick response @davidweterings @mvantellingen! I had the `TF_LOG=1` set but not the `CTP_DEBUG`. Will give it a try!

**Comment by mvantellingen** [comment]
@HenryKeen any update on this? Or can we close the issue?

**Comment by Jmen** [comment]
@mvantellingen yes, we're still getting this issue, but haven't yet dug into it to get some more info

We'll be picking this up again on Monday, and can give a proper update then

**Comment by Jmen** [comment]
We've tracked this issue down to this error, so we are investigating how the client and secret are getting created 

`Post https://api.sphere.io/moonpig-prod/extensions: oauth2: cannot fetch token: 400 Bad Request
 Response: {"statusCode":400,"message":"Permissions exceeded","errors":[{"code":"invalid_scope","message":"Permissions exceeded"}],"error":"invalid_scope","error_description":"Permissions exceeded"}`

**Comment by Jmen** [comment]
I've notice something about the creation of our api client which I'm not sure about

I found this in message in our CI logs when deploying the api client

`An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ commercetools_api_client.commerce_order_apiextension_client
      name:  "API Client for moonpig-commerce-order-apiextension" => "API Client for moonpig-commerce-extension-order"
      scope: "view_products:moonpig-prod" => "view_products:moonpig-prod manage_extensions:moonpig-prod"`

From what I understand api clients can not be modified after creation. If this is correct, then should this not be a delete and create, rather than a modify of the resource?

**Comment by davidweterings** [comment]
@Jmen we've noticed that issue too, before I continue diving into this I'd like to know what version of the Terraform provider you're using? (We've not seen it occur in the latest version).

**Comment by Jmen** [comment]
we're using 0.5.3, so we'll update and try that again

**Comment by Jmen** [comment]
We've updated to the latest version, but we've noticed an issue with the naming of the plugin. It doesn't seem to have the version number appended to the end of the file anymore e.g. terraform-provider-<NAME>_vX.Y.Z

https://www.terraform.io/docs/configuration/providers.html#plugin-names-and-versions

so terraform doesn't find the plugin when running

**Comment by davidweterings** [comment]
@Jmen I've released 0.15.1 which adds back the version number in the release archive, please try again.

**Comment by onur9** [comment]
Are you still getting this error? I hope not 😅 

Shall we close the ticket?
